specification/cognitiveservices/data-plane/FormRecognizer/preview/v1.0/FormRecognizer.json

```diff
@@ -0,0 +1,637 @@
+            "required": true,
+            "type": "string",
+            "format": "uuid"
+          }
+        ],
+        "responses": {
+          "204": {
+            "description": "Successfully removed model artifacts."
+          },
+          "default": {
+            "description": "Response entity accompanying non-successful responses containing additional details about the error.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Get Model from id": {
+            "$ref": "./examples/DeleteModelById.json"
+          }
+        }
+      }
+    },
+    "/custom/models/{id}/analyze": {
+      "post": {
+        "summary": "Analyze Form",
+        "description": "The document to analyze must be of a supported content type - 'application/pdf', 'image/jpeg' or 'image/png'. The response contains not just the extracted information of the analyzed form but also information about content that was not extracted along with a reason.",
+        "operationId": "AnalyzeWithCustomModel",
+        "consumes": [
+          "application/pdf",
+          "image/jpeg",
+          "image/png",
+          "multipart/form-data"
+        ],
+        "produces": [
+          "application/json",
+          "text/json"
+        ],
+        "parameters": [
+          {
+            "name": "id",
+            "in": "path",
+            "description": "Model Identifier to analyze the document with.",
+            "required": true,
+            "type": "string",
+            "format": "uuid"
+          },
+          {
+            "name": "keys",
+            "in": "query",
+            "description": "An optional list of known keys to extract the values for.",
+            "required": false,
+            "type": "array",
+            "items": {
+              "type": "string"
+            },
+            "uniqueItems": false
+          },
+          {
+            "name": "form_stream",
+            "in": "formData",
+            "description": "A pdf document or image (jpg,png) file to analyze.",
+            "required": true,
+            "type": "file"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/AnalyzeResult"
+            }
+          },
+          "default": {
+            "description": "Response entity accompanying non-successful responses containing additional details about the error.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Get Model from id": {
+            "$ref": "./examples/Analyze.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "TrainRequest": {
+      "description": "Contract to initiate a train request.",
+      "required": [
```

Comment:
Is this really the only required property across all model types (I can't find a "required" array in any of the other model definitions).

Reply:
So we don't have model type per se. Based on the customer input ( "source" = "SaS for Azure Blob storage with training data") we train a instance of a model we have and save the related metadata which call can refer to using modelId when calling Analyze with real data. Hope this clarifies.